tests/integration/test_mysql_protocol/test.py

```diff
@@ -790,7 +792,11 @@ def test_mysqljs_client(started_cluster, nodejs_container):
     assert code == 1
 
 
-def test_java_client(started_cluster, java_container):
+@pytest.mark.skip(
+    """Deprecated in favor of test_java_client as the scope of the tests is higher,
+    and it covers both binary and test protocols"""
+)
+def test_java_client_old(started_cluster, java_container):
```

Comment:
Maybe we can remove this, the related reference file, and the related [Java file](https://github.com/slvrtrn/ClickHouse/blob/fix-mysql-datetime-incompatibilities/docker/test/integration/mysql_java_client/Test.java) altogether, as I am not sure if it is worth keeping it as the new set of tests is much larger in terms of coverage for all MySQL protocol variations.

Reply:
Removed.